src/main/kotlin/org/rust/ide/inspections/RustSuspiciousAssignmentInspection.kt

```diff
@@ -0,0 +1,79 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import org.rust.ide.inspections.fixes.SubstituteTextFix
+import org.rust.lang.core.psi.*
+
+/**
+ * Checks for use of the non-existent =*, =! and =- operators that are probably typos but can be compiled.
+ * Analogue of Clippy's suspicious_assignment_formatting.
+ * QuickFix 1: Change `a =? b` to `a ?= b`
+ * QuickFix 2: Change `a =? b` to `a = ?b`
+ */
+class RustSuspiciousAssignmentInspection : RustLocalInspectionTool() {
+    private object Constants {
+        val LONG_TEXT_THRESHOLD = 10
+        val LONG_TEXT_SUBST = ".."
+    }
+
+    override fun getDisplayName(): String = "Suspicious assignment"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustElementVisitor() {
+            override fun visitBinaryExpr(expr: RustBinaryExprElement) {
+                if (expr.operator.text == "=") {
```

Comment:
>having only one return per function

This is actually a rather interesting aspect of Kotlin code style. I really do prefer early returns for handling "unusual" conditions. However inside lambdas `return` must be qualified ( `xs.map { return@map 92 }` ) which is not pretty, so with lambdas I usually stick with a single expression body.